### PR TITLE
test-login: skip consistency checks when logind is not active

### PR DESCRIPTION
**Pull request by keszybz**
There are two ways in swich sd_login_* functions acquire data:
some are derived from the cgroup path, but others use the data serialized
by logind.

When the tests are executed under Fedora's mock, without systemd-spawn
but instead in a traditional chroot, test-login gets confused:
the "outside" cgroup path is visible, so sd_pid_get_unit() and
sd_pid_get_session() work, but sd_session_is_active() and other functions
that need logind data fail.

Such a buildroot setup is fairly bad, but it can be encountered in the wild, so
let's just skip the tests in that case.

/* Information printed is from the live system */
sd_pid_get_unit(0, …) → "session-237.scope"
sd_pid_get_user_unit(0, …) → "n/a"
sd_pid_get_slice(0, …) → "user-1000.slice"
sd_pid_get_session(0, …) → "237"
sd_pid_get_owner_uid(0, …) → 1000
sd_pid_get_cgroup(0, …) → "/user.slice/user-1000.slice/session-237.scope"
sd_uid_get_display(1000, …) → "(null)"
sd_uid_get_sessions(1000, …) → [0] ""
sd_uid_get_seats(1000, …) → [0] ""
Assertion 'r >= 0' failed at src/libsystemd/sd-login/test-login.c:104, function test_login(). Aborting.

### CODE REVIEW

File: src/libsystemd/sd-login/test-login.c [hunk 1]
```diff
@@ -112,68 +112,74 @@ static void test_login(void) {
 
         if (session) {
                 r = sd_session_is_active(session);
-                assert_se(r >= 0);
-                log_info("sd_session_is_active(\"%s\") → %s", session, yes_no(r));
+                if (r == -ENXIO)
+                        log_notice("sd_session_is_active failed with ENXIO, it seems logind is not running.");
```
Comment (yuwata) [inline src/libsystemd/sd-login/test-login.c]: I prefer to add `()` to the function name.

**Review by yuwata — APPROVED** [review]
LGTM. One minor comment.

**Comment by keszybz** [comment]
Updated.

**Comment by keszybz** [comment]
I'll go ahead and merge this without waiting for the slow CI. It's only relaxing the check, so it's very unlikely to cause something to fail.

**Comment by poettering** [comment]
shouldn't we rather just skip the whole test if `running_in_chroot() > 0` ?

**Comment by keszybz** [comment]
> shouldn't we rather just skip the whole test if `running_in_chroot() > 0` ?

I think it's good to test that stuff returns *some* sensible answer, doesn't leak memory, etc. This patch essentially allows ENXIO as a valid answer, and that's much narrow than just skipping the tests.

BTW, I figured out why it was running in chroot:
> New Mock release v2.8
> Just a quick heads-up, there is a new set of Bodhi updates with Mock 2.8 for                                          
rather important isolation regression, no matter configuration - mock 2.7 always                                      
used --isolation=simple.  Otherwise it is small release.  Please upgrade.

**Pull request merged**